schemas/v3.1/schema.yaml

```
@@ -0,0 +1,911 @@
+      termsOfService:
+        type: string
+      contact:
+        $ref: '#/$defs/contact'
+      license:
+        $ref: '#/$defs/license'
+      version:
+        type: string
+    required:
+      - title
+      - version
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+
+  contact:
+    type: object
+    properties:
+      name:
+        type: string
+      url:
+        type: string
+      email:
+        type: string
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+
+  license:
+    type: object
+    properties:
+      name:
+        type: string
+      identifier:
+        type: string
+      url:
+        $ref: '#/$defs/uri'
+    required:
+      - name
+    oneOf:
+      - required:
+        - identifier
+      - required:
+        - url
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+
+  server:
+    type: object
+    properties:
+      url:
+        $ref: '#/$defs/uri'
+      description:
+        type: string
+      variables:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/server-variable'
+    required:
+      - url
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+
+  server-variable:
+    type: object
+    properties:
+      enum:
+        type: array
+        items:
+          type: string
+        minItems: 1
+      default:
+        type: string
+      descriptions:
+        type: string
+    required:
+      - default
+    $ref: '#/$defs/specification-extensions'
+    unevaluatedProperties: false
+
+  components:
+    type: object
+    properties:
+      schemas:
+        type: object
+        additionalProperties:
+          $dynamicRef: '#meta'
+      responses:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/response-or-reference'
+      parameters:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/parameter-or-reference'
+      examples:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/example-or-reference'
+      requestBodies:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/request-body-or-reference'
+      headers:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/header-or-reference'
+      securitySchemes:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/security-scheme-or-reference'
+      links:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/link-or-reference'
+      callbacks:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/callbacks-or-reference'
+      pathItems:
+        type: object
+        additionalProperties:
+          $ref: '#/$defs/path-item-or-reference'
+    patternProperties:
+      '^schemas|responses|parameters|examples|requestBodies|headers|securitySchemes|links|callbacks|pathItems$':
```

```suggestion
      '^(schemas|responses|parameters|examples|requestBodies|headers|securitySchemes|links|callbacks|pathItems)$':
```

without this, the ^ and $ anchors are only used on the first and last clauses respectively